### DISCUSSION TITLE
Ability to use Vue()/composition APIs in extensions

### DISCUSSION THREAD

**Discussion by sabinehoi** [directus/directus#4843] [Feature Requests]
Right now to use the `Vue()` method or vue composition APIs  you need to import all of `vue` or `composition-api` manually in the extension's package.json, which dramatically increases the extension's size (and it also takes like 5s to rollup build...). Also it is a double-import since we already use composition API's in Directus core!

This is going to be a hard to solve micro optimisation problem, but I think is necessary - we don't want people with dozens of extensions re-importing all of Vue and all of composition APIs into each extension - that will immediately slow down the app many times over

Tangentially related: #4834 - I really believe extensions should be more "first class citizens". They're just too convenient to pass up on! Happy to assist with this too following some discussion on how that can happen

**Comment by rijkvanzanten** [comment]
We should look into making that Vue object and the subsequent imports a global of the app, so extensions don't have to bundle it in somehow 🤔 (also, using regular Vue 2 object syntax doesn't require any imports)

**Comment by nickrum** [comment]
This has been addressed as part of the Vue 3 migration.
